app/controllers/api/volume_mappings_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  class VolumeMappingsController < BaseController
+    def refresh_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" if id.blank?
+
+      ensure_resource_exists(type, id) if single_resource?
+
+      api_action(type, id) do |klass|
+        volume_mapping = resource_search(id, type, klass)
+        api_log_info("Refreshing #{volume_mapping_ident(volume_mapping)}")
+        refresh_volume_mapping(volume_mapping)
+      end
+    end
+
+    def create_resource(_type, _id = nil, data = {})
+      ext_management_system = ExtManagementSystem.find(data['ems_id'])
+      task_id = VolumeMapping.create_volume_mapping_queue(session[:userid], ext_management_system, data)
```

Comment:
@gregoryb-at-ibm We need to check supports_feature here right?

Reply:
You mean to make sure that the EMS supports volume_mappings?
Like so?
```
      raise BadRequestError "Volume Mappings not supported for #{ext_management_system.class}" unless defined?(ext_management_system.class::VolumeMapping)

```

Reply:
So you have `supports_not :create` on the `VolumeMapping` base class and `supports :create` on the Autosde subclass.  We should check that the volume mapping subclass that we're going to work on supports create

Reply:
OK, so I did it this way:
```
raise BadRequestError "Volume Mappings not supported for #{ext_management_system.class}" unless ext_management_system.class::VolumeMapping.supports_create?
```